phoenix-core-client/src/main/java/org/apache/phoenix/util/CDCUtil.java

```diff
@@ -150,4 +154,30 @@ public static boolean isBinaryType(PDataType dataType) {
                 || sqlType == Types.LONGVARBINARY
                 || dataType.getSqlType() == PDataType.VARBINARY_ENCODED_TYPE);
     }
+
+    public enum CdcStreamStatus {
+        ENABLED("ENABLED"),
+        ENABLING("ENABLING"),
+        DISABLED("DISABLED"),
+        DISABLING("DISABLING");
+
+        private final String serializedValue;
+
+        private CdcStreamStatus(String value) {
+            this.serializedValue = value;
+        }
+
+        public String getSerializedValue() {
+            return serializedValue;
+        }
+    }
+
+    public static long getCDCCreationTimestamp(PTable table) {
+        for (PTable index : table.getIndexes()) {
+            if (CDCUtil.isCDCIndex(index)) {
+                return index.getTimeStamp();
+            }
+        }
```

Comment:
This method is misleading, it is returning the timestamp of the first index it finds. Since currently we can have multiple CDC indexes, it may return different value at different times. In theory, this can make `updateStreamPartitionMetadata` reuse a previous stream name instead of assigning a new one and `dropCDC` try to update the status for a wrong stream.

I am thinking, we can do the following as a quick workaround:
1. We can change `CDC_STREAM_NAME_FORMAT` to just "<table name>-UUID"` and avoid depending on this method in `updateStreamPartitionMetadata`.
2. In `dropCDC`, query for the name of the ENABLED/ENABLING stream to change its status, rather than constructing the name from `CDC_STREAM_NAME_FORMAT`.
3. In the ITs, we would assert on the existence of a stream of specific status rather than on the specific name.

Reply:
@haridsv As we discussed I added the check to only allow a single (enable/enabling) stream in createCDC. This also has the effect of only allowing a single CDC index since I throw a SQLException.

Reply:
Please correct me if I am wrong but we want to move towards single stream and single CDC index, right?

Reply:
Yes you are right, since we are not allowing more than 1 CDC, the logic is harmless, sorry about the spam.